File: JS/basic.js
/*
========================================
 JAVASCRIPT BASIC SYNTAX REFERENCE FILE
========================================
*/

// Read input (Node.js)
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split(/\s+/);
let idx = 0;

// Function example
function add(a, b) {
  return a + b;
}

// Output
console.log("Hello from JavaScript");

// Input: single value
let x = Number(input[idx++]);
console.log("x =", x);

// If-else
if (x % 2 === 0) {
  console.log("Even");
} else {
  console.log("Odd");
}

// For loop
for (let i = 0; i < 3; i++) {
  process.stdout.write(i + " ");
}
console.log();

// While loop
let i = 0;
while (i < 3) {
  process.stdout.write(i + " ");
  i++;
}
console.log();

// Array
let n = Number(input[idx++]);
let arr = [];
for (let i = 0; i < n; i++) {
  arr.push(Number(input[idx++]));
}

for (let v of arr) {
  process.stdout.write(v + " ");
}
console.log();

// Function call
console.log("Sum =", add(2, 3));

File: Python/basic.py
"""
========================================
 PYTHON BASIC SYNTAX REFERENCE FILE
========================================
"""

# Function example
def add(a, b):
    return a + b

# Output
print("Hello from Python")

# Input: single value
x = int(input())
print("x =", x)

# If-else
if x % 2 == 0:
    print("Even")
else:
    print("Odd")

# For loop
for i in range(3):
    print(i, end=" ")
print()

# While loop
i = 0
while i < 3:
    print(i, end=" ")
    i += 1
print()

# Array (list)
n = int(input())
arr = list(map(int, input().split()))

for val in arr:
    print(val, end=" ")
print()

# Function call
print("Sum =", add(2, 3))
